### PR TITLE
Juju 1678 add jammy to juju qa test

### PR DESCRIPTION
**Pull request by jameinel**
This is based of of https://github.com/juju/juju/pull/14508 with some changes to README and some slightly increased build instructions (since you actually have to build different versions of the charm for different channels).

I did actually release the charm versions, but I don't have the time to finish the PR description. Will follow up later.

## Checklist

*If an item is not applicable, use `~strikethrough~`.*

- [ ] Code style: imports ordered, good names, simple structure, etc
- [ ] Comments saying why design decisions were made
- [ ] Go unit tests, with comments saying what you're testing
- [ ] [Integration tests](https://github.com/juju/juju/tree/develop/tests), with comments saying what you're testing
- [ ] [doc.go](https://discourse.charmhub.io/t/readme-in-packages/451) added or updated in changed packages

## QA steps

*Commands to run to verify that the change works.*

```sh
QA steps here
```

## Documentation changes

*How it affects user workflow (CLI or API). Delete section if not applicable.*

## Bug reference

*Link to Launchpad bug. Delete section if not applicable.*

### CODE REVIEW

File: testcharms/charm-hub/charms/juju-qa-test/README.md [hunk 1]
```diff
@@ -6,46 +6,63 @@ A container-based V2 metadata charm to use in testing juju.
...
-|Version  |Channel          |Series				 
-|----      |----            |----               
-| 11|latest/stable|bionic, xenial|
-| 14|latest/candidate|focal, bionic, xenial|
-| 20|latest/edge|groovy, focal, bionic, xenial|
-|8   |2.0/stable  |disco, bionic, xenial, trusty |
-|10      |2.0/edge   |disco, bionic, xenial, trusty   |
+| Version    | Revision | Channel          | Series                               |
+| ---------- | -------- | ---------------- | ------------------------------------ |
+| 1.1-stable | 19       | latest/stable    | focal, bionic, xenial                |
```
Comment (hmlanigan) [inline testcharms/charm-hub/charms/juju-qa-test/README.md:22]: Let's used a different version system. A revision of a charm can be in multiple channels, and we should do that. Naming with a risk could create confusion

**Review by hmlanigan — APPROVED** [review]
Newly uploaded charm is letting CI revision tests and others pass now.

**Review by hmlanigan — APPROVED** [review]

File: testcharms/charm-hub/charms/juju-qa-test/.jujuignore [hunk 2]
```diff
@@ -4,3 +4,10 @@
 charmcraft.yaml
 build
 foo-file.txt
+edge/
+edge/version
```
Comment (jameinel) [inline testcharms/charm-hub/charms/juju-qa-test/.jujuignore:8]: I don't think you need to specify these individually if the directory itself is ignored, do you?

File: testcharms/charm-hub/charms/juju-qa-test/edge/charmcraft.yaml [hunk 3]
```diff
@@ -1,4 +1,16 @@
 type: charm
+parts:
+  charm:
+    prime:
```
Comment (jameinel) [inline testcharms/charm-hub/charms/juju-qa-test/edge/charmcraft.yaml:4]: do we need 'prime' as well as .jujuignore?

**Comment by jameinel** [comment]
Closing in favor of the 2.9 targeted version: https://github.com/juju/juju/pull/14579

**Pull request closed without merging**